common/analysis/lint_waiver_test.cc

```diff
@@ -620,5 +620,158 @@ TEST_F(LintWaiverBuilderTest, FromTextStructureOneWaiverRangeOpened) {
   EXPECT_TRUE(lint_waiver.RuleIsWaivedOnLine("qq-rule", 3));
 }
 
+TEST_F(LintWaiverBuilderTest, ApplyExternalWaiversInvalidCases) {
+  std::set<absl::string_view> active_rules;
+  const absl::string_view filename = "filename";
+
+  // Completely invalud config
+  const absl::string_view cfg_inv =
+    "inv config";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_inv));
+
+  // Valid command, inv parameters
+  const absl::string_view cfg_inv_params =
+    "waive --something";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_inv_params));
+
+  // Non-registered rule name
+  const absl::string_view cfg_inv_rule =
+    "waive --rule=abc --line=1";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_inv_rule));
+
+  // register rule
+  const absl::string_view abc_rule = "abc";
+  active_rules.insert(abc_rule);
+
+  // Valid rule, missing params
+  const absl::string_view cfg_no_param =
+    "waive --rule=abc";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_no_param));
+
+  // Valid rule, inv line number
+  const absl::string_view cfg_inv_lineno =
+    "waive --rule=abc --line=0";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_inv_lineno));
+
+  // Valid rule, inv line range
+  const absl::string_view cfg_inv_range =
+    "waive --rule=abc --line=1:0";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_inv_range));
+
+  // Valid rule, both regex and lines specified
+  const absl::string_view cfg_conflict =
+    "waive --rule=abc --regex=\".*\" --line=1";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_conflict));
+
+  // Missing rulename
+  const absl::string_view cfg_no_rule =
+    "waive --line=1";
+  EXPECT_FALSE(ApplyExternalWaivers(active_rules, filename, cfg_no_rule));
+}
+
+TEST_F(LintWaiverBuilderTest, ApplyExternalWaiversValidCases) {
+  std::set<absl::string_view> active_rules;
+  const absl::string_view filename = "filename";
+
+  // Add a single rule for tests
+  const absl::string_view abc_rule = "abc";
+  active_rules.insert(abc_rule);
+
```

Comment:
Initialize `active_rules` with this `abc_rule` instead of inserting it separately.  (This would also allow it to be declared `const`.)

Same where applicable below.